docs/ops/deployment/kubernetes.md

```diff
@@ -76,57 +74,92 @@ You can then access the Flink UI via different ways:
     2. Run `kubectl get svc flink-jobmanager-rest` to know the `node-port` of this service and navigate to [http://&lt;public-node-ip&gt;:&lt;node-port&gt;](http://<public-node-ip>:<node-port>) in your browser.
-    3. Similarly to `port-forward` solution, you could also use the following command below to submit jobs to the cluster:
+    3. If you use minikube, you can get its public ip by running `minikube ip`.
+    4. Similarly to `port-forward` solution, you could also use the following command below to submit jobs to the cluster:
 
         {% highlight bash %}./bin/flink run -m <public-node-ip>:<node-port> ./examples/streaming/WordCount.jar{% endhighlight %}
 
-In order to terminate the Flink session cluster, use `kubectl`:
+In order to terminate the Flink cluster, terminate the specific [Session](#deploy-session-cluster) or [Job](#deploy-job-cluster) cluster components
+and use `kubectl` to terminate the common components:
 
-    kubectl delete -f jobmanager-deployment.yaml
-    kubectl delete -f taskmanager-deployment.yaml
+```sh
     kubectl delete -f jobmanager-service.yaml
     kubectl delete -f flink-configuration-configmap.yaml
+    # if created then also the rest service
+    kubectl delete -f jobmanager-rest-service.yaml
+```
+
+### Deploy Session Cluster
+
+A Flink Session cluster is executed as a long-running Kubernetes Deployment.
+Note that you can run multiple Flink jobs on a Session cluster.
+Each job needs to be submitted to the cluster after the cluster has been deployed.
 
-## Flink job cluster on Kubernetes
+A basic Flink Session cluster deployment in Kubernetes has three components:
 
-A Flink job cluster is a dedicated cluster which runs a single job.
-The job is part of the image and, thus, there is no extra job submission needed.
+* a Deployment which runs a Flink Master
+* a Deployment for a pool of TaskManagers
+* a Service exposing the Flink Master's REST and UI ports
 
-### Creating the job-specific image
+After creating [the common cluster components](#deploy-flink-cluster-on-kubernetes), use [the Session specific resource definitions](#session-cluster-resource-definitions)
+to launch the Session cluster with the `kubectl` command:
 
-The Flink job cluster image needs to contain the user code jars of the job for which the cluster is started.
-Therefore, one needs to build a dedicated container image for every job.
-Please follow these [instructions](https://github.com/apache/flink/blob/{{ site.github_branch }}/flink-container/docker/README.md) to build the Docker image.
+```sh
+    kubectl create -f jobmanager-session-deployment.yaml
+    kubectl create -f taskmanager-session-deployment.yaml
+```
 
-### Using plugins
+To terminate the Session cluster, these components can be deleted before [the common ones](#deploy-flink-cluster-on-kubernetes) with the `kubectl` command:
```

Comment:
_can_ they be deleted before the common components, or _must_ they be deleted beforehand?

Reply:
Asaik, in general you create and remove resources unordered in k8s.
I can change this:
`before` -> `along with`